src/handlers/rentals.ts

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable @typescript-eslint/ban-types */
+/* eslint-disable prefer-const */
+import { log } from '@graphprotocol/graph-ts'
+import { BigInt } from '@graphprotocol/graph-ts'
+import { buildCountForAllRentals, buildCountForRentals } from '../modules/count'
+import { Rental } from '../entities/schema'
+import { OperatorUpdated, RentalStarted, AssetClaimed } from '../entities/Rentals/Rentals'
+
+function buildRentalId(contractAddress: string, tokenId: BigInt, index: BigInt): string {
+  return contractAddress + ':' + tokenId.toString() + ':' + index.toString()
+}
+
+export function handleRentalStart(event: RentalStarted): void {
+  let contractAddress = event.params._contractAddress.toHexString()
+  let tokenId = event.params._tokenId
+
+  let rentalIndex = buildCountForRentals(contractAddress, tokenId)
+  let newRentalId = buildRentalId(contractAddress, tokenId, rentalIndex.value)
+
+  let rental = new Rental(newRentalId)
+  rental.contractAddress = contractAddress
+  rental.tokenId = tokenId
+  rental.lessor = event.params._lessor.toHexString()
+  rental.tenant = event.params._tenant.toHexString()
+  rental.operator = event.params._operator.toHexString()
+  rental.rentalDays = event.params._rentalDays
+  rental.pricePerDay = event.params._pricePerDay
+  rental.sender = event.params._sender.toHexString()
+  rental.ownerHasClaimedAsset = false
+  rental.startedAt = event.block.timestamp
```

Comment:
What do you think of an `endDate` field that has the timestamp when the rental ends. Could open the possibility to query for rentals that will be finishing "soon" (in the next 7 days for example. query today < n < 7 days from today)
Can be calculated with the startedAt + rentalDays x 86400 (seconds in a day)

Reply:
That's a good idea, I'll add it.